pandas/io/formats/style.py

```diff
@@ -159,7 +159,7 @@ def __init__(
         self.index = data.index
         self.columns = data.columns
 
-        self.uuid = uuid
+        self.uuid = uuid or (uuid4().hex[:5] + "_")
```

Comment:
not averse, but what's the reason for this change?

Reply:
a) the uuid is created but later in the process, so **I changed** to bring this forward to instantiation for use with tooltips.
b) uuid1 derives using network address, so **I changed** to use uuid4 which doesn't so is potentially more network secure.
c) a resonable use case is transmitting styled dataframes to web. For large tables without setting `cell_ids` to false the uuid is transmitted at least once and potentially twice for every cell. This restricts it to 5 characters as opposed to many more so compresses the data transfer. The underscore improves HTML readability. I think the loss of randomness is a fair tradeoff?

Reply:
I reverted this. Will add it in another PR including tests.